subsys/bluetooth/audio/ascs.c

```diff
@@ -73,8 +73,32 @@ K_MEM_SLAB_DEFINE(ase_slab, sizeof(struct bt_ascs_ase),
 		  __alignof__(struct bt_ascs_ase));
 
 static struct bt_ascs sessions[CONFIG_BT_MAX_CONN];
+NET_BUF_SIMPLE_DEFINE_STATIC(ase_buf, CONFIG_BT_L2CAP_TX_MTU);
 
 static int control_point_notify(struct bt_conn *conn, const void *data, uint16_t len);
+static int ascs_ep_get_status(struct bt_audio_ep *ep,
+			      struct net_buf_simple *buf);
+
+static struct bt_ascs_ase *ascs_ase_lookup_ep(const struct bt_audio_ep *ep)
+{
+	for (size_t i = 0; i < ARRAY_SIZE(sessions); i++) {
+		struct bt_ascs *ascs = &sessions[i];
+		struct bt_ascs_ase *ase;
+		sys_snode_t *node;
+
+		SYS_SLIST_FOR_EACH_NODE(&ascs->ases, node) {
+			ase = CONTAINER_OF(node, struct bt_ascs_ase, node);
+
+			if (ep == &ase->ep) {
```

Comment:
hmm I'm not that much familiar with this code, but this & suggests that ase is container of ep,  so this function can be called with ep from some other "pool"?

Reply:
Good catch. I did not put much thought while rebasing the code. Looks like I can simply use `CONTAINER_OF` to get the ASE object